src/Evolution/Systems/NewtonianEuler/BoundaryCorrections/Hll.cpp

```diff
@@ -0,0 +1,294 @@
+    }
+  }
+  dot_product(packaged_normal_dot_flux_energy_density, flux_energy_density,
+              normal_covector);
+
+  // What value to return here..?
+  // ...
+}
+
+template <size_t Dim>
+void Hll<Dim>::dg_boundary_terms(
+    const gsl::not_null<Scalar<DataVector>*> boundary_correction_mass_density,
+    const gsl::not_null<tnsr::I<DataVector, Dim, Frame::Inertial>*>
+        boundary_correction_momentum_density,
+    const gsl::not_null<Scalar<DataVector>*> boundary_correction_energy_density,
+    const Scalar<DataVector>& mass_density_int,
+    const tnsr::I<DataVector, Dim, Frame::Inertial>& momentum_density_int,
+    const Scalar<DataVector>& energy_density_int,
+    const Scalar<DataVector>& normal_dot_flux_mass_density_int,
+    const tnsr::I<DataVector, Dim, Frame::Inertial>&
+        normal_dot_flux_momentum_density_int,
+    const Scalar<DataVector>& normal_dot_flux_energy_density_int,
+    const Scalar<DataVector>& largest_outgoing_char_speed_int,
+    const Scalar<DataVector>& largest_ingoing_char_speed_int,
+    const Scalar<DataVector>& mass_density_ext,
+    const tnsr::I<DataVector, Dim, Frame::Inertial>& momentum_density_ext,
+    const Scalar<DataVector>& energy_density_ext,
+    const Scalar<DataVector>& normal_dot_flux_mass_density_ext,
+    const tnsr::I<DataVector, Dim, Frame::Inertial>&
+        normal_dot_flux_momentum_density_ext,
+    const Scalar<DataVector>& normal_dot_flux_energy_density_ext,
+    const Scalar<DataVector>& largest_outgoing_char_speed_ext,
+    const Scalar<DataVector>& largest_ingoing_char_speed_ext,
+    const dg::Formulation dg_formulation) const noexcept {
+  // Allocate a temp buffer with four tags.
+  Variables<tmpl::list<::Tags::TempScalar<0>, ::Tags::TempScalar<1>,
+                       ::Tags::TempScalar<2>, ::Tags::TempScalar<3>>>
+      temps{get(mass_density_int).size()};
+
+  // Determine lambda_max and lambda_min from the characteristic speeds info
+  // from interior and exterior
+  get(get<::Tags::TempScalar<0>>(temps)) =
+      max(0., get(largest_outgoing_char_speed_int),
+          -get(largest_ingoing_char_speed_ext));
+  const DataVector& lambda_max = get(get<::Tags::TempScalar<0>>(temps));
+  get(get<::Tags::TempScalar<1>>(temps)) =
+      min(0., get(largest_ingoing_char_speed_int),
+          -get(largest_outgoing_char_speed_ext));
+  const DataVector& lambda_min = get(get<::Tags::TempScalar<1>>(temps));
+
+  // Pre-compute two expressions made out of lambda_max and lambda_min, for
+  // speeding up evaluating HLL flux
+  get(get<::Tags::TempScalar<2>>(temps)) = lambda_max * lambda_min;
+  const DataVector& lambdas_product = get(get<::Tags::TempScalar<2>>(temps));
+  get(get<::Tags::TempScalar<3>>(temps)) = 1. / (lambda_max - lambda_min);
+  const DataVector& one_over_lambda_max_minus_min =
+      get(get<::Tags::TempScalar<3>>(temps));
+
+  if (dg_formulation == dg::Formulation::WeakInertial) {
+    //  weak formulation
```

Comment:
Either expand on this comment, or remove it, I think. It seems redundant with the `if (dg_formulation == WeakInertial)`, no?

Reply:
Sure